### PR TITLE
Create pkg/util/http module

### PR DESCRIPTION
**Pull request by dineshg13**
### What does this PR do?

Converts `pkg/util/http` into a go module . 

### Motivation
[RFC](https://docs.google.com/document/d/1uKWa9rMNm0U-euEzEJ69KW7Id5u1E4GfYHyzC1SzYoI/edit#heading=h.wnkz5u7vx1fb)

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by pgimalac — APPROVED** [review]

**Review by davidor — APPROVED** [review]
:+1: for container-integrations (we only own 1 file)

File: pkg/logs/client/http/destination.go [hunk 1]
```diff
@@ -89,8 +90,8 @@ func NewDestination(endpoint config.Endpoint,
 	destinationsContext *client.DestinationsContext,
 	maxConcurrentBackgroundSends int,
 	shouldRetry bool,
-	telemetryName string) *Destination {
-
+	telemetryName string,
+) *Destination {
```
Comment (vickenty) [inline pkg/logs/client/http/destination.go:94]: Would it be possible to avoid reformatting the code you did not change?

**Review by amenasria — APPROVED** [review]
LGTM for Agent Platform owned file !

**Review by chouquette — APPROVED** [review]
LGTM for agent-platform file

File: pkg/logs/client/http/destination.go (hunk 1 above)
Comment (dineshg13) [reply pkg/logs/client/http/destination.go:94]: Most of the changes are coming from my editor's formatting. These changes did seem reasonable, so I left it. Let me know if they seem too concerning.

**Review by paulcacheux — APPROVED** [review]

File: pkg/logs/client/http/destination.go (hunk 1 above)
Comment (mellon85) [reply pkg/logs/client/http/destination.go:94]: `git add -p` would help as you can choose which hunks to include in the commit

Comment (dineshg13) [reply pkg/logs/client/http/destination.go:94]: That is good to know. I wasn't aware of it. Most of these changes adhere to good practices and seem reasonable. Are we good to leave them, or should I revert them?

Comment (vickenty) [reply pkg/logs/client/http/destination.go:94]: > Let me know if they seem too concerning.

Changes like this make reviewers review more code, and it becomes harder to find actual changes when using `git log` and `git blame` later. And we don't all use the same editor and editor settings. If everyone reformats everything to suit their style every time they touch a file, our git history would become much less useful.

If possible, it is better to configure your editor to leave unmodified code as is, or, as @mellon85 mentioned already, be selective in what you include in the PR.

Comment (dineshg13) [reply pkg/logs/client/http/destination.go]: Thanks. I have updated the PR by removing the un-needed changes.

**Review by hghotra — APPROVED** [review]
✅  - Serverless

**Review by vickenty — APPROVED** [review]

**Review by billyac — APPROVED** [review]

**Review by knusbaum — APPROVED** [review]

**Review by mfpierre — APPROVED** [review]

**Review by mellon85 — APPROVED** [review]

**Pull request merged**